### DISCUSSION TITLE
Managing around 500M embeddings in single AWS instance with etcd, message & object storage as S3

### DISCUSSION THREAD

**Discussion by senthil-bungee** [milvus-io/milvus#35604] [Q&A and General discussion]
Hello,

We have requirement to store all embeddings of 500M products to search in batch offline mode and not real time search. Hence we have an instance of r6a.32 box in AWS and installed standalone Milvus DB. The embeddings are generated by another application ans stored them in to S3 as parquet files. We will shutdown the DB whenever it is not used. When we get batch request to find the products, we have ECS in place to start this instance & shutdown when it is not used.

1. Each parquet file contains around 500K embeddings with 512 dim. Should I convert each embeddings into JSON for bulk insertion? What is the best way to insert them without much manipulation on the parquet files? 
3. I looked into https://milvus.io/docs/install_standalone-docker.md but some samples with value of S3 would be helpful.

**Comment by xiaofan-luan** [comment]
so it's 128core 1T node right. 
What kind of index you want to use? i you are using diskann or mmap or IVFSQ, it seems you could be able to loading everything into the clsuter. 

You can do bulkinsertion. each time you should bukinsert a batch of parquet file into milvus. But with this machine I thought the index calculation will take long(Maybe tens of hours), does that reasonable for your use case?

You should start from https://milvus.io/docs/install_standalone-docker-compose.md. with docker compose you can simply change minio related config to store data in S3.   Here is the configuration list about minio https://milvus.io/docs/system_configuration.md

It also will take a long time to load all the index before your search. What is an estimation? Is it fine to wait for a couple of minutes before you want to use it each time.
